File: iosApp/iosApp/content/Course.swift
//
// Created by Tim Ortel on 27.09.22.
//

import Foundation

/**
 * Representation of a single course.
 */
struct Course: Codable, Identifiable {
    let id: Int
    var title: String? = ""
    var description: String? = ""
    var courseIcon: String? = nil
    var semester: String? = ""
    var registrationConfirmationMessage: String? = ""
}

File: appCommon/src/commonMain/kotlin/de/tum/informatics/www1/artemis/native_app/content/Course.kt
package de.tum.informatics.www1.artemis.native_app.content

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Representation of a single course.
 */
@Serializable
data class Course(
    val id: Int,
    val title: String,
    val description: String,
    @SerialName("courseIcon") val courseIconPath: String
)